pkg/compactor/block_upload.go

```diff
@@ -262,22 +269,51 @@ func (c *MultitenantCompactor) UploadBlockFile(w http.ResponseWriter, r *http.Re
 	w.WriteHeader(http.StatusOK)
 }
 
-func (c *MultitenantCompactor) completeBlockUpload(ctx context.Context, logger log.Logger, userBkt objstore.Bucket, blockID ulid.ULID, meta metadata.Meta) error {
+func (c *MultitenantCompactor) completeBlockUpload(ctx context.Context, logger log.Logger, userBkt objstore.Bucket, blockID ulid.ULID, meta metadata.Meta) {
 	level.Debug(logger).Log("msg", "completing block upload", "files", len(meta.Thanos.Files))
 
+	// start go routine that updates validation file timestamp once per minute
+	ch := make(chan string)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	ctx, cancel := context.WithCancel(ctx)
+	go c.periodicValidationUpdater(ctx, logger, blockID, userBkt, ch, &wg, cancel)
+
+	if err := c.validateBlock(ctx, blockID, userBkt, meta); err != nil {
+		err := c.uploadValidationWithError(ctx, blockID, userBkt, err.Error())
```

Comment:
Let's also log the error returned by `validateBlock`.

Let's NOT upload validation with error IF `errors.Is(err, context.Canceled)`. If we get "context canceled" error it means we haven't finished validation until the end, so we should not mark block upload as "failed". We could delete the validation file instead (better) or just leave it as-is so that it becomes stale.